py_feels/binary/time_series.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Jan 20 14:50:21 2018
+
+Pulls feels and tweet times from the bit_feels database and performs
+a time-window average of the sentiments for plotting. pushes means,
+standard devaitions, classifier names and bin times into the 'stats'
+table of bit_feels. Takes two arguments: window and env
+
+window is the width of the time-bin in hours,
+env is "dev" or "prod"
+
+@author: rcehemann
+"""
+
+import pandas as pd
+from numpy import linspace, digitize, mean, std
+from datetime import datetime
+from sys import argv
+from sqlalchemy import create_engine
+
+# check environment, parse arguments
+if len(argv) < 2:
+    window = 48.
+    env = "bit_feels_dev"
+elif len(argv) < 3 or argv[2] == "dev":
+    window = argv[1]
+    env = "bit_feels_dev"
+elif argv[2] == "prod":
+    window = argv[1]
+    env = "bit_feels"
+else:
+    raise EnvironmentError("second argument should be 'dev' or 'prod'")
+
+# set up SQL connection
+engine = create_engine('postgresql+psycopg2://wojak:@localhost/' + env)
+
+def ts_to_td(inserted_at):
+    """
+        returns the total elapsed seconds from
+        the timestamp to present time.
+    """
+    return (inserted_at - datetime.now()).total_seconds()
+
+def time_window(times, window=24.):
+    """
+        function to partition a time series
+        into time windows. takes a numerical value
+        in HOURS for the requested time window.
+        bins are statically sized and the number of bins
+        is calcluated based on the earliest timestamp in the db
+    """
+    t_min  = min(times.values)
+    t_max  = max(times.values)
+    n_bins = int((t_max - t_min) / 3600 / window)
+    bins   = linspace(t_min, t_max, num=n_bins)
+    windows = digitize(times, bins, right=True)
+    bins = [datetime.fromtimestamp(t) for t in bins]
+    print(len(bins))
+    return windows, bins
+
+def mean_and_std(times):
+    return mean(times), std(times)
+
+query  = "SELECT tweet_id, sentiment, classifier FROM feels"
+feels  = pd.read_sql(query, engine)
+query  = "SELECT inserted_at, tweet_id FROM tweets"
```

Comment:
`inserted_at` is the time that we inserted the tweet into the database. I think `created_at` might make more sense here.

Reply:
hodor